src/storage/src/hummock/sstable/multi_builder.rs

```diff
@@ -290,7 +295,25 @@ where
                     .num_pending_write_io
                     .fetch_add(1, std::sync::atomic::Ordering::SeqCst);
             }
-            let builder = self.builder_factory.open_builder().await?;
+            let mut builder = self.builder_factory.open_builder().await?;
+            if add_point_delete_epoch != HummockEpoch::MAX {
+                // We do not hope the left-bound of key-range of a sstable be `Excluded`, so we
+                // split the last range to be be too part. For example, [a, +∞) and
+                // (b, f) will be cut into [a, b), b, (b,f) three range, and [a,b) belongs to the
+                // last file. we must add b to the next file. Since all point key
+                // less than `event` have been add to builder before, there will be no point-key
+                // larger than `event-key`.
+                builder
+                    .add(
+                        FullKey::from_user_key(
+                            event.event_key.left_user_key.as_ref(),
+                            add_point_delete_epoch,
```

Comment:
Is is possible that there is another KV with same user key but larger epoch added later to the new builder, which breaks the order of keys.